nixos/modules/services/x11/xserver.nix

```diff
@@ -246,7 +246,11 @@ in
       videoDrivers = mkOption {
         type = types.listOf types.str;
         # !!! We'd like "nv" here, but it segfaults the X server.
-        default = [ "radeon" "cirrus" "vesa" "vmware" "modesetting" ];
+        default =
+          [ "radeon" "cirrus" "vesa" ]
+          ++ lib.optional (pkgs.stdenv.hostPlatform.isx86_32 || pkgs.stdenv.hostPlatform.isx86_64) "vmware"
```

Comment:
Don't we need parentheses around the whole lib.optional thing?

Reply:
this can also be simplified with `pkgs.stdenv.hostPlatform.isx86` - I doubt `x86_128-linux` will come any time soon ;-)

Reply:
I went for the better option of check `meta` on the driver derivation.